protocol/manager.go

```diff
@@ -57,10 +59,24 @@ func (m *Manager) IBDPeer() *peerpkg.Peer {
 
 // AddTransaction adds transaction to the mempool and propagates it.
 func (m *Manager) AddTransaction(tx *util.Tx) error {
-	return m.context.AddTransaction(tx)
+	err := m.context.AddTransaction(tx)
+	if err != nil {
+		if protocolErr := &(protocolerrors.ProtocolError{}); errors.As(err, &protocolErr) {
+			return err
+		}
+		panic(err)
+	}
+	return nil
 }
 
 // AddBlock adds the given block to the DAG and propagates it.
 func (m *Manager) AddBlock(block *util.Block, flags blockdag.BehaviorFlags) error {
-	return m.context.AddBlock(block, flags)
+	err := m.context.AddBlock(block, flags)
+	if err != nil {
+		if protocolErr := &(protocolerrors.ProtocolError{}); errors.As(err, &protocolErr) {
+			return err
+		}
+		panic(err)
```

Comment:
Please add "TODO(libp2p): Remove panic once RPC is integrated into protocol architecture"